### DISCUSSION TITLE
전체적인 이 프로젝트의 문제점들을 적어보기 - 토미

### DISCUSSION THREAD

**Discussion by yewon-yw** [familymoments/family-moments-android#173] [Ideas]
1. 패키지 정리하기
2. api response가 fail 일 때 뷰모델에서 Throwable 메시지로 확인하고 있는데 변경 필요
3. 이미지 관련 로직들 정리 필요해보임
    - 압축, 리사이징 등
4. 공통 UI 따로 컴포넌트 만들어서 사용하기
    - FMButton이나 FMTextField 만들어놨는데 사용하지 못하고 직접 만들어서 쓰는 경우가 많음
5. API 중복 호출 막기
    - Modifier clickable 관련 확장함수 전체적으로 적용하기
6. 네트워크 연결 끊겼을 때 팝업 필요
7. sharedPreference 대신 DataStore로 변경 필요

**Comment by boring-km** [comment]
1. 패키지 정리하기는 app 모듈에 있는 build.gradle에 있는 의존성들 정리인가요?
2. 어떻게 변경하면 좋을지도 같이 생각해볼까요?
3. 정리라는게 리팩토링이랑 기능 개선이 같이 포함되는 거겠죠..?
4. 공통 UI는 디자인에서 어느 정도 공통된 컴포넌트로 만들어줘야 실천이 용이할 거 같긴해요
7. 왜 SharedPreference 대신 DataStore로 변경이 필요한지도 같이 적어주면 좋을 것 같아요..!

**Reply by yewon-yw** [reply]
1. 패키지가 너무 많아서 프로가드가 제대로 적용되는지에 대한 의문을 가졌던 기억이 나서 작성한 것이었는데, 구조를 새로 짤 예정이라 그럴 필요가 없을 것 같네요😮
2. 베리님이 작성해 주신 것처럼 오류 타입을 만들어놓고 사용하는 방법이 좋아 보여요!
3. 리사이징 로직 자체가 블로그에서 가져온 내용이어서 어떻게 동작하는지에 대한 이해 없이 적용되어 있습니다🥲 그래서 코드를 다시 작성해야 한다고 느꼈습니다! 그리고 이미지 관련 로직을 FileUtil에 모아놓기보다는 조금 더 파일을 세분화해서 가독성을 높이는 방향을 생각하고 작성했습니다!
4. 회의에서 대신 전달해 주셔서 감사합니다! 😁
5. dataStore의 가장 큰 장점으로는 flow기반이기 때문에 비동기적으로 동작한다는 점인 것 같아요! sharedPreferences는 UI 스레드를 차단시킬 가능성이 있지만 dataStore는 안전하다는점.. 다른 차이도 있는데 글로 설명하는 것보단 사진으로 보는 게 편할 것 같아서 첨부해놔용
<img src = "https://github.com/user-attachments/assets/dbedc28a-e729-4634-b905-e378458e9325" width = 500>


[이미지 참고](https://developer.android.com/codelabs/android-preferences-datastore?hl=ko#3) 
[sharedPreferences 단점](https://developer.android.com/reference/kotlin/androidx/datastore/core/DataStore?hl=en&_gl=1*cn19h7*_up*MQ..*_ga*Njk4NjY2MzkzLjE3Mjg5MjE1NDI.*_ga_6HH9YJMN9M*MTcyODkyMTU0MS4xLjAuMTcyODkyMTU0MS4wLjAuMjAwODA0MzMzNw..)

**Reply by boring-km** [reply]
1번 -> 프로가드 적용되는지 확인하는 건 debug 빌드에서 난독화를 적용해놓고, 프로가드 파일까지 적용했을 때 정상적으로 동작하는지 확인하면 하나하나 확인해 보기 편합니다..!
